src/api/Metadata.js

```diff
@@ -719,6 +723,72 @@ class Metadata extends File {
             this.errorHandler(e);
         }
     }
+
+    /**
+     * API URL for metadata suggestions
+     *
+     * @return {string} url for metadata suggestions
+     */
+    getMetadataSuggestionsUrl(): string {
+        return `${this.getBaseApiUrl()}/metadata_instances/suggestions`;
+    }
+
+    /**
+     * Gets suggestions for possible metadata key-value pairs for the given item
+     *
+     * @param {string} id - Id of the item to pull metadata from
+     * @param {string} type - Type of item. Only "file” is supported.
+     * @param {string} scope - The metadata template scope
+     * @param {string} templateKey - The metadata template key to apply
+     * @param {string} confidence - The confidence level the suggestion must surpass. Only “experimental” is supported.
+     * @return {Array<MetadataSuggestion>} array of metadata templates
+     */
+    async getMetadataSuggestions(
+        id: string,
+        type: typeof TYPE_FILE,
+        scope: string,
+        templateKey: string,
+        confidence: typeof METADATA_SUGGESTIONS_CONFIDENCE_EXPERIMENTAL,
+    ): Promise<Array<MetadataSuggestion>> {
+        this.errorCode = ERROR_CODE_FETCH_METADATA_SUGGESTIONS;
+
+        if (!id || type !== TYPE_FILE) {
+            throw getBadItemError();
+        }
+
+        if (!scope) {
+            throw new Error('Missing scope');
+        }
+
+        if (!templateKey) {
+            throw new Error('Missing templateKey');
+        }
+
+        if (confidence !== METADATA_SUGGESTIONS_CONFIDENCE_EXPERIMENTAL) {
+            throw new Error(`Invalid confidence level: "${confidence}"`);
+        }
+
+        const url = this.getMetadataSuggestionsUrl();
+        let suggestions = {};
+        try {
+            suggestions = await this.xhr.get({
+                url,
+                id: getTypedFileId(id),
+                params: {
+                    item: `${type}_${id}`,
+                    scope,
+                    template_key: templateKey,
+                    confidence,
+                },
+            });
+        } catch (e) {
+            const { status } = e;
+            if (isUserCorrectableError(status)) {
+                throw e;
```

Comment:
I think most other API methods invoke an error handler rather than re-throwing. Can we verify we're handling these errors in the conventional way ?

Reply:
But maybe it should accept `errorCallback` function, similar to `getMetadata`. 
The goal is to trigger `errorCallback` with correct error code when AI Suggestion endpoint fail, so the consumer application can handle it eg. with a banner or with custom logger.

Reply:
Followed up on Slack with @wpiesiak. This implementation is similar to `getTemplates` and `getInstances`.